sdk/schemaregistry/azure-data-schemaregistry/src/main/java/com/azure/data/schemaregistry/SchemaRegistryClient.java

```diff
@@ -122,8 +143,27 @@ public SchemaProperties registerSchema(String groupName, String name, String sch
     @ServiceMethod(returns = ReturnType.SINGLE)
     public Response<SchemaProperties> registerSchemaWithResponse(String groupName, String name, String schemaDefinition,
         SchemaFormat format, Context context) {
-        return this.asyncClient.registerSchemaWithResponse(groupName, name, schemaDefinition, format,
-            context).block();
+        if (Objects.isNull(groupName)) {
+            throw logger.logExceptionAsError(new NullPointerException("'groupName' should not be null."));
+        } else if (Objects.isNull(name)) {
+            throw logger.logExceptionAsError(new NullPointerException("'name' should not be null."));
+        } else if (Objects.isNull(schemaDefinition)) {
+            throw logger.logExceptionAsError(new NullPointerException("'schemaDefinition' should not be null."));
+        } else if (Objects.isNull(format)) {
+            throw logger.logExceptionAsError(new NullPointerException("'format' should not be null."));
+        }
+
+        logger.verbose("Registering schema. Group: '{}', name: '{}', serialization type: '{}', payload: '{}'",
+            groupName, name, format, schemaDefinition);
+
+        context = enableSyncRestProxy(context);
```

Comment:
Do we need to set this for every method? Can this be changed to a policy instead and set at the client level?

Reply:
Since it applies to all sync APIs in this case, that's doable I believe.

Reply:
This cannot be done at policy level, as this info is read at Rest Proxy level through the Context. 
Policies come into play post Rest Proxy I believe.
Also, this change is temporary and will be removed later on once we enable sync stack by default across all SDKs.